api.go

```diff
@@ -2,7 +2,9 @@ package vshard_router
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"reflect"
```

Comment:
reflect usage sounds scary, cause it is very slow library, please describe why u used it, instead default libraries